src/networking.c

```diff
@@ -1267,12 +1267,9 @@ void unlinkClient(client *c) {
     }
 
     /* Remove from the list of pending reads if needed. */
-    if (c->flags & CLIENT_PENDING_READ) {
```

Comment:
Right. Note that the previous code was alse inefficient because we still search in the pending reads list. I decided to remove this for now and do O(n) just to make sure we can safely know if we're in an IO thread context without relying on flags. I'll improve this later. Good point.